src/ProjectService.Business/Commands/ProjectUsers/FindProjectUsersCommand.cs

```diff
@@ -22,71 +19,75 @@ namespace LT.DigitalOffice.ProjectService.Business.Commands.ProjectUsers
 {
   public class FindProjectUsersCommand : IFindProjectUsersCommand
   {
-    private readonly IBaseFindFilterValidator _baseFindFilterValidator;
     private readonly IProjectUserRepository _projectUserRepository;
     private readonly IUserService _userService;
     private readonly IImageService _imageService;
     private readonly IPositionService _positionService;
     private readonly IUserInfoMapper _userInfoMapper;
 
     public FindProjectUsersCommand(
-      IBaseFindFilterValidator baseFindFilterValidator,
       IProjectUserRepository projectUserRepository,
       IUserService userService,
       IImageService imageService,
       IPositionService positionService,
       IUserInfoMapper userInfoMapper)
     {
-      _baseFindFilterValidator = baseFindFilterValidator;
       _projectUserRepository = projectUserRepository;
       _userService = userService;
       _imageService = imageService;
       _positionService = positionService;
       _userInfoMapper = userInfoMapper;
     }
 
-    public async Task<FindResultResponse<UserInfo>> ExecuteAsync(Guid projectId, FindProjectUsersFilter filter)
+    public async Task<FindResultResponse<UserInfo>> ExecuteAsync(
+      Guid projectId, FindProjectUsersFilter filter, CancellationToken cancellationToken)
     {
-      if (!_baseFindFilterValidator.ValidateCustom(filter, out List<string> errors))
+      List<DbProjectUser> projectUsers = await _projectUserRepository.GetAsync(
+        projectId: projectId, isActive: filter.IsActive, cancellationToken: cancellationToken);
+
+      List<string> errors = new();
+
+      if (projectUsers is null || !projectUsers.Any())
       {
-        return ResponseCreatorStatic.CreateFindResponse<UserInfo>(statusCode: HttpStatusCode.BadRequest, errors: errors);
+        return new(errors: errors);
       }
 
-      List<DbProjectUser> projectUsers = await _projectUserRepository.GetAsync(projectId: projectId, isActive: filter.IsActive);
+      IEnumerable<Guid> usersIds = projectUsers.Select(pu => pu.UserId);
 
-      if (projectUsers is null)
+      if (filter.ByPositionId.HasValue)
       {
-        return new();
+        PositionFilteredData data = (await _positionService.GetPositionFilteredDataAsync(
+          new List<Guid>() { filter.ByPositionId.Value }, errors))?.FirstOrDefault();
+
+        usersIds = data is not null ? usersIds.Where(i => data.UsersIds.Contains(i)).ToList() : Enumerable.Empty<Guid>();
```

Comment:
May be "usersIds.Where(data.UsersIds.Contains)"?

Reply:
done